modules/kubernetes/ingress-nginx/main.tf

```diff
@@ -29,6 +29,133 @@ resource "kubernetes_namespace" "this" {
   }
 }
 
+resource "kubernetes_network_policy" "deny_default" {
+  metadata {
+    name      = "deny-default"
+    namespace = kubernetes_namespace.this.metadata[0].name
+  }
+
+  spec {
+    pod_selector {
+      match_labels = {}
+    }
+
+    policy_types = ["Ingress", "Egress"]
+
+    ingress {
+      from {
+        pod_selector {}
+      }
+    }
+
+    egress {
+      to {
+        namespace_selector {}
+        pod_selector {
+          match_labels = {
+            k8s-app = "kube-dns"
+          }
+        }
+      }
+
+      ports {
+        port     = 53
+        protocol = "UDP"
+      }
+    }
+
+    egress {
+      to {
+        pod_selector {}
+      }
+    }
+  }
+}
+
+resource "kubernetes_network_policy" "allow_scraping" {
+  metadata {
+    name      = "allow-scraping"
+    namespace = kubernetes_namespace.this.metadata[0].name
+  }
+
+  spec {
+    pod_selector {
+      match_labels = {
+        "app.kubernetes.io/component" = "controller"
+        "app.kubernetes.io/instance" = "ingress-nginx"
+        "app.kubernetes.io/name" = "ingress-nginx"
+      }
+    }
+
+    policy_types = ["Ingress"]
+
+    ingress {
+      from {
+        namespace_selector {
+          match_labels = {
+            name = "prometheus"
+          }
+        }
+        pod_selector {
+          match_labels = {
+            app = "prometheus"
+            "app.kubernetes.io/name" = "prometheus"
+          }
+        }
+      }
+
+      ports {
+        port = "metrics"
+        protocol = "TCP"
+      }
+    }
+  }
+}
+
+resource "kubernetes_network_policy" "allow_ingress_traffic" {
+  metadata {
+    name      = "allow-ingress-traffic"
```

Comment:
Are we sure network policies are applied to LB services? Is it the same in all clouds or different?

Reply:
Network policies have no effect on services, only on pods. What we are doing is allowing traffic originating from IPs outside the cluster, to reach the nginx pods.